kclvm/sema/src/resolver/node.rs

```diff
@@ -185,6 +185,51 @@ impl<'ctx> MutSelfTypedResultWalker<'ctx> for Resolver<'ctx> {
                             self.ctx.ty_ctx.infer_to_variable_type(value_ty.clone()),
                         );
                     }
+                    // If the assign statement has a type annotation like: `a: str = "test_str"`,
+                } else if assign_stmt.type_annotation.is_some() {
+                    // Get the type annotation.
+                    if let Some(ty_annotation) = &assign_stmt.ty {
+                        let annotation_ty = self
+                            .parse_ty_with_scope(&ty_annotation.node, ty_annotation.get_span_pos());
+                        // Set the type annotation as the target type.
+                        self.set_type_to_scope(name, annotation_ty.clone(), target.get_span_pos());
```

Comment:
Split these logic to a smaller function.

Reply:
fixed